### PR TITLE
fix(yarn): yarn v2 yarnrc support

### PR DESCRIPTION
**Pull request by JamieMagee**
Part of #5230 

Still doesn't work without manually specifying the yarn compatability (that is coming in a separate PR)

### CODE REVIEW

File: lib/manager/npm/post-update/index.ts [hunk 1]
```diff
@@ -157,12 +157,15 @@ export async function writeExistingFiles(
...
+        content: yarnrcContent,
+      } = packageFile.yarnrc;
+      logger.debug(`Writing ${yarnrcFilename} to ${basedir}`);
       try {
         await outputFile(
           yarnrcFilename,
-          packageFile.yarnrc
+          yarnrcContent
             .replace('--install.pure-lockfile true', '')
             .replace('--install.frozen-lockfile true', '')
```
Comment (viceice) [inline lib/manager/npm/post-update/index.ts:170]: Are you sure this works for the new yaml file? I think we need different config options for yarn v2

**Review by viceice — COMMENTED** [review]
needs test coverage too

File: lib/manager/npm/post-update/index.ts (hunk 1 above)
Comment (JamieMagee) [reply lib/manager/npm/post-update/index.ts:170]: Yes, it doesn't work. But if the flag doesn't exist in the file, then there is nothing to replace.

File: lib/manager/npm/extract/yarn.ts [hunk 2]
```diff
@@ -41,3 +46,29 @@ export async function getYarnLock(
...
+export async function getYarnRc(
+  packageFilePath: string,
+  config: ExtractConfig
+): Promise<RcFile> {
+  if (is.string(config.yarnrc) || config.localDir === undefined) {
+    return undefined;
+  }
+  const yarnRcFileNames = [
+    '.yarnrc',
+    '.yarnrc.yml',
```
Comment (ylemkimon) [inline lib/manager/npm/extract/yarn.ts:59]: `.yarnrc.yml` overrides (takes precedence over) `.yarnrc` in Yarn 1 (https://git.io/JUcco, https://github.com/renovatebot/renovate/pull/7220#discussion_r485383364).

**Pull request closed without merging**